include/bpf2c.h

```diff
@@ -2,6 +2,9 @@
 // SPDX-License-Identifier: MIT
 
 #pragma once
+
+#include <intrin.h>
```

Comment:
Couple of queries:

1. Why are we using this header (from clang) when we have windows implementation for the ```Interlocked...``` family in ```winnt.h```?
2. Why is the header being included here? Given its specificity to the ```bpf_code_generator.cpp``` source file, maybe it could be included directly in that source itself?  That would localize the header dependency to the source(s) where it's actually used.

Reply:
If we want atomic function header only accurs when the file call those functions, we may need a flag to indicate
if atomic function was used, change like this:

```cpp
// bpf_code_generator.cpp
void
bpf_code_generator::encode_instructions(const bpf_code_generator::unsafe_string& section_name)
{
// [...]
            if ((inst.opcode & EBPF_MODE_MASK) == EBPF_MODE_ATOMIC) { // MODE_ATOMIC
                need_atomic = true; // set flag
                auto line = std::string("");
// [...]
void
bpf_code_generator::emit_c_code(std::ostream& output_stream)
{
    // Emit C file
    output_stream << "#include \"bpf2c.h\"" << std::endl << std::endl;
    if (need_atomic) {
        output_stream << "#include <intrin.h>" << std::endl << std::endl;
    }
```

And I just found those atomic functions are also defined in `wdm.h`, so for `*_sys.c`, we do not need to include another header. If we can tolerate include intrin.h in bpf2c.h, we can just add intrin.h in bpf2c.h like this:

```cpp
// bpf2c.h
#if defined(NO_CRT)
//[...]
#else
#include <stdbool.h>
#include <stddef.h>
#include <stdint.h>
#include <intrin.h>
#endif
```

Which way is better?

Reply:
That's the thing :-) What's confusing me why we even _need_ ```intrin.h``` when we have native windows equivalents (via native windows headers).  I'm quite possibly missing something here. In any case it looks like this involves some 'policy' issues so I'd like to hear what @Alan-Jowett / @dthaler have to say about this.

We can discuss this issue during our Monday morning meeting.

Reply:
Can we get a sample that uses these instructions added first? I would like to see what the generated code looks like before I comment.